.unreleased/LLT-5277

```diff
@@ -0,0 +1 @@
+Extract socket protection mechanism from `telio-sockets` into a separate `sock-prot` crate for reuse by other libs
```

Comment:
The changelog is mostly there for apps integrating libtelio, and this is a very under-the-hood change, so I might be wrong but I don't think it needs a changelog entry

Reply:
Ok, I'll remove it

Reply:
Done